src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.PlatformVersion.cs

```diff
@@ -0,0 +1,94 @@
+            => IsOSPlatform(osPlatform) && IsOSVersionOrLater(major, minor, int.MinValue, int.MinValue);
+
+        public static bool IsOSPlatformOrLater(OSPlatform osPlatform, int major, int minor, int build)
+            => IsOSPlatform(osPlatform) && IsOSVersionOrLater(major, minor, build, int.MinValue);
+
+        public static bool IsOSPlatformOrLater(OSPlatform osPlatform, int major, int minor, int build, int revision)
+            => IsOSPlatform(osPlatform) && IsOSVersionOrLater(major, minor, build, revision);
+
+        public static bool IsOSPlatformEarlierThan(string platformName)
+        {
+            (OSPlatform platform, Version version) = Parse(platformName);
+
+            return IsOSPlatformEarlierThan(platform, version.Major, version.Minor, version.Build, version.Revision);
+        }
+
+        public static bool IsOSPlatformEarlierThan(OSPlatform osPlatform, int major)
+            => IsOSPlatform(osPlatform) && Environment.OSVersion.Version.Major < major;
+
+        public static bool IsOSPlatformEarlierThan(OSPlatform osPlatform, int major, int minor)
+            => IsOSPlatform(osPlatform) && !IsOSVersionOrLater(major, minor, int.MinValue, int.MinValue);
+
+        public static bool IsOSPlatformEarlierThan(OSPlatform osPlatform, int major, int minor, int build)
+            => IsOSPlatform(osPlatform) && !IsOSVersionOrLater(major, minor, build, int.MinValue);
+
+        public static bool IsOSPlatformEarlierThan(OSPlatform osPlatform, int major, int minor, int build, int revision)
+            => IsOSPlatform(osPlatform) && !IsOSVersionOrLater(major, minor, build, revision);
+
+        private static bool IsOSVersionOrLater(int major, int minor, int build, int revision)
+        {
+            Version current = Environment.OSVersion.Version;
+            if (current.Major != major)
+            {
+                return current.Major > major;
+            }
+            if (current.Minor != minor)
+            {
+                return current.Minor > minor;
+            }
+            if (current.Build != build)
+            {
+                return current.Build > build;
+            }
+
+            return current.Revision >= revision;
+        }
+
+        private static (OSPlatform, Version) Parse(string platformName)
+        {
+            if (platformName == null)
+            {
+                throw new ArgumentNullException(nameof(platformName));
+            }
+            if (platformName.Length == 0)
+            {
+                throw new ArgumentException(SR.Argument_EmptyValue, nameof(platformName));
+            }
+
+            for (int i = 0; i < platformName.Length; i++)
+            {
+                if (char.IsDigit(platformName[i]))
+                {
+                    if (Version.TryParse(platformName.AsSpan().Slice(i), out var version))
+                    {
+                        return (OSPlatform.Create(platformName.Substring(0, i)), version);
```

Comment:
Should the platform name be trimmed?  Wondering how permissive we want to be on inputs here.

Reply:
I meant what if someone writes:
"Linux 1.2.3"
Currently this will call Create with "Linux " rather than "Linux", and that will fail to match OSPlatform.Linux.  I'm wondering what behavior we want.

Reply:
Ah--actual _trimming_. 😄 

Hmmm... I can't think of any objective reasons for either trimming or not. Unless there's a good reason to be permissive here, I would suggest not to do trimming for the sake of simplicity.

Reply:
> I would suggest not to do trimming for the sake of simplicity

👍